src/tpu_manager.rs

```diff
@@ -27,7 +27,7 @@ pub struct TpuManager {
     error_count: Arc<AtomicU32>,
     rpc_client: Arc<RpcClient>,
     // why arc twice / one is so that we clone rwlock and other so that we can clone tpu client
-    tpu_client: Arc<RwLock<Arc<QuicTpuClient>>>,
```

Comment:
I think you dont understand what i was doing here. First thing is that you are exposing RwLockReadGuard<QuicTpuClient> as pub which is a big mistake. I have explicitly added comment why there are two arc. With your changes when the connection is being reseted it has to wait for all the other read locks to finish reading to reset the connection. That is not we want, when we reset the connection it will directly update the TPUClient.

Please dont do things that you dont understand.